addons/purchase/views/product_views.xml

```diff
@@ -13,10 +13,40 @@
+                                <field name="sequence" widget="handle"/>
+                                <field name="product_tmpl_id" invisible="1"/>
+                                <field name="name" context="{'default_customer': 0, 'search_default_supplier': 1, 'default_supplier': 1}" optional="show"/>
+                                <field name="date_start" optional="hide"/>
+                                <field name="date_end" optional="hide"/>
+                                <field name="product_id" groups="product.group_product_variant" domain="[('product_tmpl_id', '=', product_tmpl_id)]" options="{'no_create_edit': True}" optional="hide"/>
+                                <field name="product_name" optional="hide"/>
+                                <field name="product_code" optional="hide"/>
+                                <field name="delay" string="Lead Time" optional="hide"/>
+                                <field name="company_id" options="{'no_create': True}" optional="hide"/>
+                                <field name="min_qty" optional="hide"/>
+                                <field name="price" optional="show"/>
+                            </tree>
+                        </field>
                     </group>
                     <group string="Vendor Prices" name="variant_vendors" attrs="{'invisible': [('product_variant_count','&lt;=',1)]}">
-                        <field name="variant_seller_ids" context="{'default_product_tmpl_id': context.get('product_tmpl_id', active_id)}" nolabel="1"/>
+                        <field name="variant_seller_ids" context="{'default_product_tmpl_id': context.get('product_tmpl_id', active_id)}" nolabel="1">
+                            <tree editable="bottom">
+                                <field name="sequence" widget="handle"/>
+                                <field name="product_tmpl_id" invisible="1"/>
+                                <field name="name" context="{'default_customer': 0, 'search_default_supplier': 1, 'default_supplier': 1}" optional="show"/>
+                                <field name="date_start" optional="hide"/>
+                                <field name="date_end" optional="hide"/>
+                                <field name="product_id" groups="product.group_product_variant" domain="[('product_tmpl_id', '=', product_tmpl_id)]" options="{'no_create_edit': True}" optional="hide"/>
+                                <field name="product_name" optional="hide"/>
+                                <field name="product_code" optional="hide"/>
+                                <field name="delay" string="Lead Time" optional="hide"/>
+                                <field name="company_id" options="{'no_create': True}" optional="hide"/>
+                                <field name="min_qty" optional="hide"/>
+                                <field name="price" optional="show"/>
+                            </tree>
+                        </field>
```

Comment:
As you have two time the exact same arch, you can create a record for your list and then pass its `id` in the field  using the `tree_view_ref` in the context.